frontend/lib/src/components/elements/CodeBlock/StreamlitSyntaxHighlighter.test.tsx

```diff
@@ -47,10 +47,10 @@ describe("CustomCodeTag Element", () => {
     const { baseElement } = render(<StreamlitSyntaxHighlighter {...props} />)
 
     expect(baseElement.querySelector("pre code")?.outerHTML).toBe(
-      '<code class="language-plaintext" style="white-space: pre;"><span>import streamlit as st\n' +
-        "</span>\n" +
-        'st.write("Hello")\n' +
-        "</code>"
+      '<code class="language-plaintext" style="white-space: pre;"><span><span>import streamlit as st\n' +
```

Comment:
do you know where these new `span` elements come from? Is it related to always overwriting the `renderer` function?

Reply:
Yes, exactly. That's what these lines do:

```
                    lineNumberElement,
                    {
                      children,
                      properties: { className: [] },
                      tagName: "span",
                      type: "element",
                    },
```

What that block does is change every line of code from this...

```
<>
  {lineNumber}
  {actualLineOfCode}
<>
```

...to this:
```
<>
  {lineNumber}
  <span>
    {actualLineOfCode}
  </span>
<>
```